contracts/validators/CredentialAtomicQueryValidator.sol

```diff
@@ -42,19 +52,44 @@ abstract contract CredentialAtomicQueryValidator is OwnableUpgradeable, ICircuit
         uint256[2] calldata a,
         uint256[2][2] calldata b,
         uint256[2] calldata c,
-        uint256 queryHash
+        bytes calldata cirquitQueryData
     ) external view virtual returns (bool) {
         // verify that zkp is valid
         require(verifier.verifyProof(a, b, c, inputs), "Proof is not valid");
+        CredentialAtomicQuery memory credAtomicQuery = abi.decode(
+            cirquitQueryData,
+            (CredentialAtomicQuery)
+        );
+
+        if (credAtomicQuery.queryHash == 0) {
```

Comment:
do not calculate hash, but use         
one of (queryHashMerklized /  queryHashNonMerklized) depending on merklized public signal

Reply:
e.g.if pubsignals[ index of merklized ] == 1 : queryHashMerklized ? queryHashNonMerklized